cli/src/lib.rs

```diff
@@ -2100,6 +2103,31 @@ fn cluster_url(cfg: &Config) -> String {
     }
 }
 
+fn clean(cfg_override: &ConfigOverride) -> Result<()> {
+    let cfg = Config::discover(cfg_override)?.expect("Not in workspace.");
+    let cfg_parent = cfg.path().parent().expect("Invalid Anchor.toml");
+    let target_dir = cfg_parent.join("target");
+    let deploy_dir = target_dir.join("deploy");
+
+    for entry in fs::read_dir(target_dir)? {
+        let path = entry?.path();
+        if path.is_dir() && path != deploy_dir {
+            fs::remove_dir_all(path).expect("Could not remove directory.");
```

Comment:
pls provide as much helpful info as possible in error msgs. In this case, include the path of the dir that could not be removed

Reply:
added a warning with the failed file/directory to be deleted and the error. I think it is better not to panic as the following files could be deleted successfully

Reply:
I actually think panicing is the better thing to do here.

if an error happens here something unusual is happening that should be investigated and also should make CI flows fail.

Reply:
yes, I hadn't thought of it that way. I have added back the error (but more verbose)